Fix $openldap::server::database data type

smortex opened the pull request:
#### Pull Request (PR) description

It is expected to be a boolean value.  Was typed as string by mistake in
c2ea6cb4f17798eb5cd39709d0b996ed6cd4b1dc.

#### This Pull Request (PR) fixes the following issues

Fixes #328

kenyon commented on manifests/server/database.pp:
```diff
@@ -7,7 +7,7 @@
   Optional[String[1]]                           $rootdn          = undef,
   Optional[String[1]]                           $rootpw          = undef,
-  Optional[String[1]]                           $initdb          = undef,
+  Optional[Boolean]                             $initdb          = undef,
```
I find `Optional` strange for Booleans, because isn't `undef` equivalent to `false`? Seems like the default is `true` in the type. Maybe remove `Optional` and default it to `true`?

smortex replied:
Yes, this is indeed over-complicated and I was trying to decipher this.
Defaulting to true should be the right thing to do.

smortex replied:
I had to reverse this because the default value actually depend on the backend.

smortex commented:
Okay, to quote a comment once read on a mailing list, "Don't trust the comments!" :-)  The parameter defaults to true… unless we are in the cases where it default to false :upside_down_face: 

At least, we have a CI that helped reading the right information in the right place!

Reverted and updated the doc to match the code.
